pycozmo/conn.py

```diff
@@ -158,19 +168,35 @@ def run(self) -> None:
             except Empty:
                 continue
 
+            raw_frames = []
+
             # Construct frame
             if isinstance(pkt, protocol_encoder.Ping):
-                frame = Frame(protocol_declaration.FrameType.PING, 0, 0, self.last_ack, [pkt])
+                raw_frames.append(
+                    Frame(protocol_declaration.FrameType.PING, 0, 0, self.last_ack, [pkt])
+                    .to_bytes())
             else:
-                seq = self.window.put(pkt)
-                if pkt.type == PacketType.COMMAND:
-                    frame = Frame(protocol_declaration.FrameType.ENGINE_ACT, seq, seq, self.last_ack, [pkt])
-                else:
-                    frame = Frame(protocol_declaration.FrameType.ENGINE, seq, seq, self.last_ack, [pkt])
-            raw_frame = frame.to_bytes()
+                first_seq, pkts = self.window.put(pkt)
+                framelen = 0
+                seq = first_seq
+                pkts = []
+                for p in pkts:
+                    framelen += len(p) + 1
+                    if framelen < 1000:
```

Comment:
We should define `MAX_FRAME_SIZE` which will probably be something like 1492 - `MIN_FRAME_SIZE` = 1478.

I'll try to check in the PCAPs that I have what is the maximum frame size, I've seen between Cozmo and the app.

Reply:
Completely agree, I put a random 1000 based on the biggest frames I've seen so far, so I'd be happy if you can find out the max size we can use.

Reply:
I checked, and the maximum frame size I've seen in PCAPs is 1420 B.

This is the total size of the UDP packet data, including the Cozmo frame header.